docs/topics/yaml-rule-structure-syntax.adoc

```diff
@@ -0,0 +1,520 @@
+
+[source,terminal]
+----
+when:
+  builtin.filecontent:
+    filePattern: "<regex_to_match_filenames_to_scope_search>"
+    pattern: "<regex_to_match_content_in_the_matching_files>"
+----
+
+.`xml`
+
+The `xml` capability enables the provider to query XPath expressions on a list of provided XML files. This capability takes 2 input parameters.
+
+[source,terminal]
+----
+when:
+  builtin.xml:
+    xpath: "<xpath_expressions>" <1>
+    filepaths: <2>
+      - "/src/file1.xml"
+      - "/src/file2.xml"
+----
+<1> `xpath` must be a valid XPath expression.
+<2> `filepaths` is a list of files to apply the XPath query to.
+
+.`json`
+
+The `json` capability enables the provider to query XPath expressions on a list of provided JSON files. Currently, `json` only takes XPath as input and performs the search on all JSON files in the codebase.
+
+[source,terminal]
+----
+when:
+  builtin.json:
+    xpath: "<xpath_expressions>" <1>
+----
+<1> `xpath` must be a valid XPath expression.
+
+.`hasTags`
+
+The `hasTags` capability enables the provider to query application tags. It queries the internal data structure to check whether the application has the given tags.
+
+[source,terminal]
+----
+when:
+  # when more than one tags are given, a logical AND is implied
+  hasTags: <1>
+    - "tag1"
+    - "tag2"
+----
+<1> When more than one tags is given, a logical AND is implied.
+
+===== `java` provider
+
+The `java` provider analyzes Java source code.  This provider's capabilities are `referenced` and `dependency`.
+
+.`referenced`
+
+The `referenced` capability enables the provider to find references in the source code. This capability takes two input parameters, `pattern` and `location`.
+
+[source,terminal]
+----
+when:
+  java.referenced:
+    pattern: "<pattern>" <1>
+    location: "<location>" <2>
+----
+<1> A RegEx pattern to match, for example, `org.kubernetes.*`
+<2> Specifies the exact location where the pattern needs to be matched, for example, `IMPORT`
+
+The supported locations are the following:
+
+* `CONSTRUCTOR_CALL`
+* `TYPE`
+* `INHERITANCE`
+* `METHOD_CALL`
+* `ANNOTATION`
+* `IMPLEMENTS_TYPE`
+* `ENUM_CONSTANT`
+* `RETURN_TYPE`
+* `IMPORT`
+* `VARIABLE_DECLARATION`
+
+.`dependency`
+
+The `dependency` capability  enables the provider to find dependencies for a given application. {ProductShortName} generates a list of the application's dependencies, and you can use this capability to query the list and check whether a certain dependency (within a version range) exists for the application.
+
+[source,terminal]
+----
+when:
+  java.dependency:
+    name: "<dependency_name>" <1>
+    upperbound: "<version_string>" <2>
+    lowerbound: "<version_string>" <3>
+----
+<1> Name of the dependency to search for
+<2> Upper bound on the version of the dependency
+<3> Lower bound on the version of the dependency
+
+===== `go` provider
+
+The `go` provider analyzes Golang source code.  This provider's capabilities are `referenced` and `dependency`.
+
```

Comment:
"Golang" does not appear to be used officially. Suggestions: "analyzes Go source code" or "analyzes source code written in Go."